src/muya/lib/utils/index.js

```diff
@@ -155,6 +157,61 @@ export const loadImage = async (url, detectContentType = false) => {
   })
 }
 
+export const isOnline = () => {
+  return navigator.onLine === true
+}
+
+export const getPageTitle = url => {
+  // No need to request the title when it's not url.
+  if (!url.startsWith('http')) {
+    return ''
+  }
+  // No need to request the title when off line.
+  if (!isOnline()) {
+    return ''
+  }
+
+  const req = new XMLHttpRequest()
+  let settle
+  const promise = new Promise((resolve, reject) => {
+    settle = resolve
+  })
+  const handler = () => {
+    if (req.readyState === XMLHttpRequest.DONE) {
+      if (req.status === 200) {
+        const contentType = req.getResponseHeader('Content-Type')
+        if (/text\/html/.test(contentType)) {
+          const { response } = req
+          if (typeof response === 'string') {
+            const match = response.match(/<title>(.*)<\/title>/)
+            return match[1] ? settle(match[1]) : settle('')
```

Comment:
@Jocs  I think we should escape the title using DOMPurify to prevent possible XSS?

Reply:
BTW we should check if the array contains at least one title (length >= 2). I don't know what JS does if the index is out of bounds.

Reply:
>BTW we should check if the array contains at least one title (length >= 2). I don't know what JS does if the index is out of bounds due missing title.

it will always have `match[1]` if the match is not `null`. 

One html page will have more than one `title`  ? as far as I can see, html will only has one title.

Reply:
There is a missing `match && match[1]` or `match && match.length >= 2` check. Otherwise an exception is thrown or segfault.

Reply:
@fxha I'll make a PR to solve later.